spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/ForwardPathFilter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2013-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.springframework.cloud.gateway.filter;
+
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.core.Ordered;
+import org.springframework.web.server.ServerWebExchange;
+
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ROUTE_ATTR;
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.isAlreadyRouted;
+
+/**
+ * Filter to set the path in the request URI if the {@link Route} URI has the scheme
+ * <code>forward</code>.
+ * @author Ryan Baxter
+ */
+public class ForwardPathFilter implements GlobalFilter, Ordered{
```

Comment:
Why another filter?

Reply:
I wanted to set the path of the request url before the other filters have a chance to manipulate it.  The current ForwardRouteFilter would be executed after all that takes place.